tests/check/check_test.go

```diff
@@ -166,11 +172,211 @@ func TestCheckLogs(t *testing.T) {
 			require.JSONEq(t, test.expectedContext["raw_response"].(string), string(fields["raw_response"].(json.RawMessage)))
 			require.Equal(t, test.expectedContext["authorization_model_id"], fields["authorization_model_id"])
 			require.Equal(t, test.expectedContext["store_id"], fields["store_id"])
+			require.Equal(t, test.expectedContext["user_agent"], fields["user_agent"])
 			require.NotEmpty(t, fields["datastore_query_count"])
 			require.NotEmpty(t, fields["peer.address"])
 			require.NotEmpty(t, fields["request_id"])
 			require.NotEmpty(t, fields["trace_id"])
-			require.Len(t, fields, 12)
+			require.Len(t, fields, 13)
+		})
+	}
+
+}
+
+func TestCheckLogsWithHTTP(t *testing.T) {
+	// create mock OTLP server
+	otlpServerPort, otlpServerPortReleaser := run.TCPRandomPort()
+	localOTLPServerURL := fmt.Sprintf("localhost:%d", otlpServerPort)
+	otlpServerPortReleaser()
+	_, serverStopFunc, err := mocks.NewMockTracingServer(otlpServerPort)
+	defer serverStopFunc()
+	require.NoError(t, err)
+
+	cfg := run.MustDefaultConfigWithRandomPorts()
+	cfg.Trace.Enabled = true
+	cfg.Trace.OTLP.Endpoint = localOTLPServerURL
+	cfg.Datastore.Engine = "memory"
+
+	observerLogger, logs := observer.New(zap.DebugLevel)
+	serverCtx := &run.ServerContext{
+		Logger: &logger.ZapLogger{
+			Logger: zap.New(observerLogger),
+		},
+	}
+
+	// We're starting a full fledged server because the logs we
+	// want to observe are emitted on the interceptors/middleware layer.
+	cancel := tests.StartServerWithContext(t, cfg, serverCtx)
+	defer cancel()
+
+	conn, err := grpc.Dial(cfg.GRPC.Addr,
```

Comment:
I don't see this used other than in the `defer conn.Close()` below.

My recommendation here would be to cut down on the HTTP boilerplate below and use the gRPC client to create the store, write the model, write the tuples, etc..

Then you use the HTTP `http.Post(...)` just for the Check call, which you are asserting produces the log that the `user_agent` field should be part of.

This will clean up the boilerplate code that is here.

Reply:
Actually, thinking it a bit more - I will move the test to the TestCheckLogs test